assets/js/src/integrations/classic.js

```diff
@@ -0,0 +1,106 @@
+import { tracker } from '../tracker';
+import { getProductFromID } from '../utils';
+import {
+	events,
+	cart,
+	products,
+	product,
+	addedToCart,
+	order,
+} from '../config.js';
+
+/**
+ * The Google Analytics integration for classic WooCommerce pages
+ * triggers events using three different methods.
+ *
+ * 1. Automatically handle events listed in the global `wcgaiData.events` object.
+ * 2. Listen for custom events from WooCommerce core.
+ * 3. Listen for various actions (i.e clicks) on specific elements.
+ */
+
+export const trackClassicIntegration = () => {
+	const eventData = {
+		storeCart: cart,
+		products,
+		product,
+		order,
+	};
+
+	Object.values( events ?? {} ).forEach( ( eventName ) => {
+		if ( eventName === 'add_to_cart' ) {
+			tracker.eventHandler( eventName )( { product: addedToCart } );
+		} else {
+			tracker.eventHandler( eventName )( eventData );
+		}
+	} );
+
+	/**
+	 * Track the custom add to cart event dispatched by WooCommerce Core
+	 *
+	 * @param {Event} e - The event object
+	 * @param {Object} fragments - An object containing fragments of the updated cart.
+	 * @param {string} cartHash - A string representing the hash of the cart after the update.
+	 * @param {HTMLElement[]} button - An array of HTML elements representing the add to cart button.
+	 */
+	document.body.onadded_to_cart = ( e, fragments, cartHash, button ) => {
+		tracker.eventHandler( 'add_to_cart' )( {
+			product: getProductFromID(
+				parseInt( button[ 0 ].dataset.product_id )
+			),
+		} );
+	};
+
+	/**
+	 * Attach click event listeners to all remove from cart links on page load and when the cart is updated.
+	 */
+	const removeFromCartListener = () => {
+		document
+			.querySelector( '.woocommerce-cart-form' )
+			?.addEventListener( 'click', ( e ) => {
+				const item = e.target.closest(
+					'.woocommerce-cart-form__cart-item .remove'
+				);
+
+				if ( ! item || ! item.dataset.product_id ) {
+					return;
+				}
```

Comment:
:question: 
Technically, we do not _"Attach click event listeners to all remove from cart links"_ we do "Attach a click event listener to the entire form".
And I wonder why? Why do we listen to all the clicks anywhere in the form, instead of 

```js
document
    .querySelectorAll( '.woocommerce-cart-form .woocommerce-cart-form__cart-item .remove[data-product_id]')
    .forEach( (item) => item.addEventListener( 'click', removeFromCartHandler ) );
    
function removeFromCartHandler( e ) {
	tracker.eventHandler( 'remove_from_cart' )( {
		product: getProductFromID(
			parseInt( e.target.dataset.product_id ),
			true
		),
	} );
}    
```
Also, having the handler as the same, not anonymous function for all `addEventListener`s, would guard us against attaching the same listener twice. 
(as mentioned in https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/328#discussion_r1473645572)

----

The suggested code would not work if the remove from cart buttons are added asynchronously after the page load and `onupdated_wc_div`